src/lib/shared.ts

```diff
@@ -9,7 +10,7 @@ export type GenericCallback<T> = (error?: Error, result?: T) => void
 
 export type VoidCallback = () => void
 
-export type IStream = Duplex & {
+export type IStream = (Duplex | NativeDuplex) & {
```

Comment:
This change actually represents the reality of how `IStream` is used. Since this is just a type, it won't affect runtime.
I added `@ts-expect-error` directives in other places so we don't break existing code.